src/mlib/int128.test.cpp

```diff
@@ -239,6 +240,42 @@ static_assert (mlib_int128_negate (9223372036854775809_i128) <
                "fail");
 #endif
 
+#ifdef __SIZEOF_INT128__
+// mlib_int128_to_native converts mlib_int128 to native __uint128_t.
+// Endian-ness is accounted for.
+static void
+mlib_int128_to_native (mlib_int128 in, __uint128_t *out)
+{
+   uint8_t *out_u8 = (uint8_t *) out;
+   if (MLIB_IS_BIG_ENDIAN) {
+      // Copy hi, then lo.
+      memcpy (out_u8, &in.r.hi, sizeof (in.r.hi));
+      memcpy (out_u8 + sizeof (in.r.hi), &in.r.lo, sizeof (in.r.lo));
+   } else {
+      // Copy lo, then hi.
+      memcpy (out_u8, &in.r.lo, sizeof (in.r.lo));
+      memcpy (out_u8 + sizeof (in.r.lo), &in.r.hi, sizeof (in.r.hi));
+   }
+}
```

Comment:
Recommend returning by-value instead of through a pointer, i.e. `static inline __uint128_t mlib_int128_to_native(mlib_int128 in)`. Same with the reverser function.